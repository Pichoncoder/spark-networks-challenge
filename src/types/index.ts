import { Data, Matches, Match } from "../../common/interfaces";

export enum FiltersProperties  {
    has_photo = "main_photo",
    has_contact = "contacts_exchanged",
    is_favourite = "favourite",
    in_compatibility_range = "compatibility_score",
    in_age_range = "age",
    in_height_range = "height_in_cm",
    in_my_location = "city_name",
  }

export interface FiltersTypes  {
    has_photo: BooleanFilters;
    is_favourite: BooleanFilters;
    has_contact: BooleanFilters;
    in_compatibility_range: NumericFilters;
    in_height_range: NumericFilters;
    in_age_range: NumericFilters;
    in_my_location: StringFilters;
}

export interface FilterTypesKey {
    [key: string]: any; // Add index signature
}

export type BooleanFilters = {
    filter: string;
    text: string;
    value: boolean,
     func: (el: IUserDetails) => boolean;
};

export type StringFilters = {
    filter: string;
    text: string;
    value: string,
    func: (el: IUserDetails) => boolean;
};

export type NumericFilters = {
    filter: string;
    text: string;
    min: number;
    max: number;
    value: Array<number>;
    func: IFilterFunction;
};

export interface IFilterFunction {
    (el: IUserDetails, value?: any): boolean;
}

export interface IFilterFunctions extends Array<IFilterFunction> {
    (el: IUserDetails): boolean;
}

export type State = {
    data: FilteredData,
    filtered_data: FilteredData,
    filters: FiltersTypes,
};

export interface FilteredData extends Array<Match> { }

export interface UserProps {
    details: Match;
}

export interface IUserDetails extends Match {}

export type IFilteringFunction = {
    (el: IUserDetails | null, filter?: string): boolean | undefined
} | IFilteringFunctionDefault;

export type IFilteringFunctionDefault = {
    (el: IUserDetails | null, filter?: string): false;
};

export interface IFilterCheckboxProps  {
    handleCheckbox: (filter: string, value: any) => void;
    filter: string;
    text: string;
}

export interface IFilterRangeProps  {
    handleRange: (filter: string, value: number[]) => void;
    min: number;
    max: number;
    filter: string;
    text: string;
    value: number[];
}

