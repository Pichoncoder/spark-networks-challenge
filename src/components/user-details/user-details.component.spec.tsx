import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TestRenderer, { create } from "react-test-renderer";
import UserDetails from "./user-details.component";
import { shallow } from "enzyme";
import { UserProps } from "../../types";

describe("<UserDetails /> spec", () => {
  it("Snapshot:check the component", () => {
    const props: UserProps = {
        details: {
            display_name: "Chuck",
            age: 33,
            job_title: "Ninja",
            height_in_cm: 170,
            city: {
              name: "New York",
              lat: 54.00,
              lon: -0.118092
            },
            compatibility_score: 3,
            contacts_exchanged: 0,
            favourite: false,
            religion: "Jew",
        }
    };

      const container = TestRenderer.create(<UserDetails {...props}></UserDetails>);
      expect(container).toMatchSnapshot();
  });
});