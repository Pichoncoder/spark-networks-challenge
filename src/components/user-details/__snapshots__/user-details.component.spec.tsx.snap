// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetails /> spec Snapshot:check the component 1`] = `
<div
  className="user-details card"
>
  <div
    className="user-details__pic"
  >
    <img
      alt="Chuck"
      className="user-details__pic-img"
      src="https://via.placeholder.com/100"
    />
  </div>
  <ul
    className="user-details__box --top-info"
  >
    <li
      className="user-details__li"
    >
      <span
        className="user-details__name"
      >
        Chuck
      </span>
    </li>
    <li
      className="user-details__li"
    >
      <span
        className="user-details__job"
      >
        Ninja
      </span>
    </li>
    <li
      className="user-details__li"
    >
      <span
        className="user-details__fav"
      />
    </li>
  </ul>
  <ul
    className="user-details__box --buttom-info"
  >
    <li
      className="user-details__li --split-box"
    >
      <span
        className="user-details__religion"
      >
        Jew
      </span>
      <span
        className="user-details__city"
      />
    </li>
    <li
      className="user-details__li --split-box"
    >
      <span
        className="user-details__age"
      >
        33
      </span>
      <br />
      <span
        className="user-details__height"
      >
        170
      </span>
    </li>
    <li
      className="user-details__li"
    >
      <span
        className="user-details__contacts"
      >
        0
      </span>
    </li>
    <li
      className="user-details__li"
    >
      <span
        className="user-details__score"
      >
        3
      </span>
    </li>
  </ul>
</div>
`;
